tests/profiler/test_profiler.py

```diff
@@ -165,6 +167,61 @@ def test_batches_are_accessible(
         == file_list
     )
 
+    batch_request_1 = BatchRequest(
+        datasource_name="generic_csv_generator",
+        data_connector_name="daily_data_connector",
+        data_asset_name="daily_data_asset",
+        data_connector_query={
+            "index": -1,
+        },
+    )
+    # Should give most recent batch
+    validator_1 = context.get_validator(
+        batch_request=batch_request_1,
+        create_expectation_suite_with_name="my_expectation_suite_name_1",
+    )
+    metric_max = validator_1.get_metric(
+        MetricConfiguration("column.max", metric_domain_kwargs={"column": "batch_num"})
+    )
+    assert metric_max == 20
+
+    batch_request_2 = BatchRequest(
+        datasource_name="generic_csv_generator",
+        data_connector_name="daily_data_connector",
+        data_asset_name="daily_data_asset",
+        data_connector_query={
+            "index": -2,
+        },
+    )
+    validator_2 = context.get_validator(
+        batch_request=batch_request_2,
+        create_expectation_suite_with_name="my_expectation_suite_name_2",
+    )
+    metric_max = validator_2.get_metric(
+        MetricConfiguration("column.max", metric_domain_kwargs={"column": "batch_num"})
+    )
+    assert metric_max == 19
+
+    for batch_num in range(1, 20 + 1):
+        batch_request = BatchRequest(
+            datasource_name="generic_csv_generator",
+            data_connector_name="daily_data_connector",
+            data_asset_name="daily_data_asset",
+            data_connector_query={
+                "index": -batch_num,
+            },
+        )
+        validator = context.get_validator(
+            batch_request=batch_request,
+            create_expectation_suite_with_name=f"my_expectation_suite_name__{batch_num}",
+        )
+        metric_max = validator.get_metric(
+            MetricConfiguration(
+                "column.max", metric_domain_kwargs={"column": "batch_num"}
+            )
+        )
+        assert metric_max == 21 - batch_num
+
```

Comment:
This is very interesting -- even though we are analyzing one batch at a time, we are still calling this multi-batch?  I was imagining that at least one multi-batch use case would involve getting several batches, and processing them.  Is this a valid thinking?  Thank you.

Reply:
Correct - this test for now is just to check that we can access all batches individually. We will definitely need to do that as well, this is sort of a baby step in that direction.